predicators/spot_utils/spot_utils.py

```diff
@@ -304,9 +320,17 @@ def get_apriltag_pose_from_camera(
                     fiducial_rt_camera_frame[2])
 
             # Get graph_nav to body frame.
-            self.graph_nav_command_line.set_initial_localization_fiducial()
-            state = self.graph_nav_command_line.graph_nav_client.\
-                get_localization_state()
+            exec_start = time.time()
+            while True:
```

Comment:
can we just have the while loop say `while time.perf_counter() - exec_start > self.localization_timeout` instead of `while True`?